<a href="https://colab.research.google.com/github/bcury/Python-Projects/blob/main/ML_Estat%C3%ADstica_DadosAbsolutos_Relativos_coeficientesTaxas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Calculos estatisticos são fundamenatais para entender os algoritimos de Machine Learning

**Dados absolutos e relativos**

In [1]:
import pandas as pd


**Percentuais**

In [15]:
dados = {'emprego':['Programador','engenheiro','Cientista'],
         'Liverpool':[97350, 82080,112840],
         'Milan':[77140, 71540, 62310],
         'Madrid':[10034,12345,12390]}

In [16]:
type(dados)

dict

In [17]:
dados

{'Liverpool': [97350, 82080, 112840],
 'Madrid': [10034, 12345, 12390],
 'Milan': [77140, 71540, 62310],
 'emprego': ['Programador', 'engenheiro', 'Cientista']}

passando o dicionario para dataframe

In [18]:
base = pd.DataFrame(dados)

In [24]:
base #As cidades referem as qts de contratações

emprego  Liverpool  Milan  Madrid
0  Programador      97350  77140   10034
1   engenheiro      82080  71540   12345
2    Cientista     112840  62310   12390

somas das colunas individualmente

In [20]:
base['Liverpool'].sum()

292270

In [21]:
base['Milan'].sum()

210990

In [22]:
base['Madrid'].sum()

34769

novas coluna agora, percentuais de cada contratações de cada cidade

In [25]:
base['% Liverpool'] = (base['Liverpool'] / base['Liverpool'].sum())*100

In [27]:
base['% Milan'] = (base['Milan'] / base['Milan'].sum())*100

In [30]:
base['%Madrid'] = (base['Madrid'] / base['Madrid'].sum())*100

In [31]:
base

emprego  Liverpool  Milan     Madrid  % Liverpool    % Milan    %Madrid
0  Programador      97350  77140  28.859041    33.308242  36.560974  28.859041
1   engenheiro      82080  71540  35.505767    28.083621  33.906820  35.505767
2    Cientista     112840  62310  35.635192    38.608136  29.532205  35.635192

Portanto calculamos os dados absolutos e relativos

**Taxas**

In [39]:
dados2 = {'ano': ['1','2','3','4','total'],
          'matricula_inicio': [70,50,44,21,185],
          'matricula_fim': [65,49,40,20,174]}

In [40]:
type(dados2)

dict

In [41]:
dados2

{'ano': ['1', '2', '3', '4', 'total'],
 'matricula_fim': [65, 49, 40, 20, 174],
 'matricula_inicio': [70, 50, 44, 21, 185]}

In [42]:
base2 = pd.DataFrame(dados2)

In [43]:
base2

ano  matricula_inicio  matricula_fim
0      1                70             65
1      2                50             49
2      3                44             40
3      4                21             20
4  total               185            174

queremos calcular a evasão, então vamos criar essa coluna

In [47]:
base2['taxa_evasao'] = ((base2['matricula_inicio'] - base2['matricula_fim']) / base2['matricula_inicio']) *100

In [48]:
base2

ano  matricula_inicio  matricula_fim  taxa_evasao
0      1                70             65     7.142857
1      2                50             49     2.000000
2      3                44             40     9.090909
3      4                21             20     4.761905
4  total               185            174     5.945946

aqui temos as taxas de evasões da escola/universidade, sendo no geral de 5.95%